wallet/src/main/scala/org/bitcoins/wallet/models/SpendingInfoDAO.scala

```diff
@@ -11,21 +16,29 @@ import org.bitcoins.core.protocol.transaction.{
 import org.bitcoins.core.wallet.utxo.TxoState
 import org.bitcoins.db.CRUDAutoInc
 import org.bitcoins.wallet.config._
-import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.ProvenShape
 
 import scala.concurrent.{ExecutionContext, Future}
 
 case class SpendingInfoDAO()(
     implicit val ec: ExecutionContext,
-    val appConfig: WalletAppConfig)
+    override val appConfig: WalletAppConfig)
     extends CRUDAutoInc[SpendingInfoDb] {
-
   import org.bitcoins.db.DbCommonsColumnMappers._
+  import profile.api._
 
   /** The table inside our database we are inserting into */
-  override val table = TableQuery[SpendingInfoTable]
-  private val addrTable = TableQuery[AddressTable]
+  override val table: slick.lifted.TableQuery[SpendingInfoTable] = profile.api.TableQuery[SpendingInfoTable]
+
+  private lazy val addrTable: slick.lifted.TableQuery[AddressDAO#AddressTable] = {
+    AddressDAO()(ec, appConfig).table
+      .asInstanceOf[TableQuery[AddressDAO#AddressTable]]
+  }
 
+  private lazy val txTable: slick.lifted.TableQuery[IncomingTransactionDAO#IncomingTransactionTable] = {
+    IncomingTransactionDAO().table
+      .asInstanceOf[TableQuery[IncomingTransactionDAO#IncomingTransactionTable]]
+  }
```

Comment:
I don't think we need these casts

Reply:
We do unfortunately, i can't remove them on any tables where we were are referencing another `DAO`